### PR TITLE
Sed 2530 ability to have multiple analytics dashboards

### PR DESCRIPTION
**Pull request by lucianbaciu**

### CODE REVIEW

File: step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package step.plugins.timeseries.dashboards.model;
+
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+public class ChartFilterItem {
```
Comment (david-stephan) [inline step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java:7]: For all classes with Chart in their name, are they restricted to chart or apply to dashblet. In the later case, they should be renamed

File: step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	
+	private String label; // optional. should be used for filter options
```
Comment (david-stephan) [inline step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java:9]: Seems this class is having optional fields depending on the filter type, shouldn't we have a class hierarchy here?

File: step-core/src/main/java/step/core/entities/EntityManager.java [hunk 3]
```diff
@@ -46,6 +46,7 @@ public class EntityManager  {
 	public final static String recursive = "recursive";
 	public final static String measurements = "measurements";
     public final static String metricTypes = "metricTypes";
```
Comment (david-stephan) [inline step-core/src/main/java/step/core/entities/EntityManager.java:48]: indentattion issue

**Review by david-stephan — CHANGES_REQUESTED** [review]

File: step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java (hunk 1 above)
Comment (lucianbaciu) [reply step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java:7]: @david-stephan they are used in dashlets, but also in the dashboard (the global filters). maybe call them TimeSeriesFilterItem?

Comment (david-stephan) [reply step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java:7]: @lucianbaciu ultimetaley global filters are used by dashlets too, but TimeSeriesFilterItem is fine with me.

File: step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java (hunk 2 above)
Comment (lucianbaciu) [reply step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/dashboards/model/ChartFilterItem.java]: @david-stephan all the attributes are together now. We can split into multiple classes based on the type, but I'm not sure it is worth it. This way the processing is very straight in both FE and BE.

**Review by david-stephan — CHANGES_REQUESTED** [review]
@lucianbaciu See the individual comment, I will directly commit changes for all of them except for the cleanup of "createTestDashboard" and extension of the junit coverage for dashboard. Please take care of that.

File: step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesControllerPlugin.java [hunk 4]
```diff
@@ -25,113 +31,244 @@
...
+	}
+
+	private void createLegacyDashboard(DashboardAccessor dashboardAccessor) {
+		boolean legacyDashboardExists = dashboardAccessor.stream().anyMatch(d -> d.getMetadata().get("isLegacy") != null && d.getMetadata().get("isLegacy").equals(true));
+		if (!legacyDashboardExists) {
+			DashboardView dashboard = new DashboardView();
+			dashboard.setName("Performance Dashboard");
+			dashboard.getMetadata().put("isLegacy", true);
+			dashboard.getMetadata().put("link", "analytics");
+			dashboardAccessor.save(dashboard);
```
[the remark this answers is not in the document]
Comment (lucianbaciu) [reply step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesControllerPlugin.java:97]: @david-stephan I don't see any edit regarding this. do you want me to do it?

[the remark this answers is not in the document]
Comment (david-stephan) [reply step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesControllerPlugin.java:97]: @lucianbaciu this is handle in the new EE plugin.

**Comment by lucianbaciu** [comment]
@david-stephan regarding test coverage, the dashboard service provides nothing more than the CRUD + your method accessor method findLegacyDashboards() which may be tested because has custom logic.
- Do we need to test the CRUD, or better make tests in the framework when we move the models there?
- If yes, how should I test the service? should I follow TimeSeriesExecutionPluginTest and create the entire contexts/plugins etc? I don't know how to inject the service otherwise.

**Comment by david-stephan** [comment]
@lucianbaciu there is no point adding unit test on the services, there is no new logic here. You're adding many classes as part of this PR with new accessor and model, make sure to have coverage for this code and definitely cleanup the unused function in the plugin class

**Pull request merged**